rtl/eclrtl/eclregex.cpp

```diff
@@ -389,16 +410,57 @@ class CCompiledStrRegExpr : implements ICompiledStrRegExpr
 // STRING implementation
 //---------------------------------------------------------------------------
 
+/**
+ * @brief Fetches or creates a compiled string regular expression object.
+ *
+ * This function fetches a compiled string regular expression object from the cache if it exists,
+ * or creates a new one if it doesn't. The regular expression object is created based on the provided
+ * regex pattern, length, and case sensitivity flag. The created object is then cached for future use.
+ *
+ * @param _regexLength      The length of the regex pattern.
+ * @param _regex            The regex pattern.
+ * @param _isCaseSensitive  Flag indicating whether the regex pattern is case sensitive or not.
+ * @return  A pointer to a copy of the fetched or created CCompiledStrRegExpr object.  The returned object
+ * *        must eventually be deleted.
+ */
+CCompiledStrRegExpr* fetchOrCreateCompiledStrRegExpr(int _regexLength, const char * _regex, bool _isCaseSensitive)
+{
+    CCompiledStrRegExpr * compiledObjPtr = nullptr;
+    uint32_t options = (_isCaseSensitive ? 0 : PCRE2_CASELESS);
+    hash64_t regexHash = HASH64_INIT;
+
+    // Create a hash of the regex pattern plus the options
+    regexHash = rtlHash64Data(_regexLength, _regex, regexHash);
+    regexHash = rtlHash64Data(sizeof(options), &options, regexHash);
+    
+    // Check the cache
+    {
+        CriticalBlock lock(compiledStrRegExprLock);
+        compiledObjPtr = dynamic_cast<CCompiledStrRegExpr*>(compiledStrRegExprCache.get(regexHash));
```

Comment:
My recollection is that a dynamic cast is significantly less efficient than a static cast, or a virtual method call.  It might be worth considering if it can be removed, but probably not worth it for this PR.

Reply:
It is less efficient, but safer; I blame muscle memory for this one.  I've swapped in static_cast because "I know what I'm doing" (hahaha).  static_cast is resolved at compile time.